### PR TITLE
Change http to https for security links

### PR DESCRIPTION
**Pull request by trungnvfet**
For an explanation of how to fill out the fields, please see the relevant section 
in [PULL_REQUESTS.md](./PULL_REQUESTS.md)

*Description*:  Fix http link to https link for secure.
*Risk Level*: Low 
*Testing*:
*Docs Changes*:
*Release Notes*:
[Optional Fixes #Issue]
[Optional *Deprecated*:]

### CODE REVIEW

**Comment by truongnh1992** [comment]
@trungnvfet  For the DCO check, when you commit your changes, you have to use `signed-off`:
```sh
git commit -s
```
If you have one commit or the latest commit in the PR is missing a `signed-off`, the simplest way to fix this is to run:
```sh
git commit --amend -s
```
then push your changes to your dev branch.

**Comment by trungnvfet** [comment]
@truongnh1992 Thank you

File: docs/root/intro/comparison.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ proxy:
   run a mix of nginx and haproxy. A single proxy solution at every hop is substantially simpler from
   an operations perspective.
 
-`haproxy <http://www.haproxy.org/>`_
```
Comment (junr03) [inline docs/root/intro/comparison.rst]: several title underlines are too short, and that is why the docs runs is erroring

Comment (trungnvfet) [reply docs/root/intro/comparison.rst]: Right. Thank you for your review.

**Pull request merged**